packages/cli/src/utils/getClientVersion.ts

```diff
@@ -33,6 +39,68 @@ async function getPrismaClientVersionFromNodeModules(cwd: string = process.cwd()
   }
 }
 
+type DriverAdapterVersion = {
+  name: OfficialDriverAdapters
+  version: string
+  underlyingDriver: {
+    name: OfficialUnderlyingDrivers
+    version: string | null
+  }
+}
+
+/**
+ * Try reading the Prisma Driver Adapter versions from the local package.json.
+ * This also includes the underlying driver versions, if found. They are looked up
+ * among the `dependencies` only, with the exception of `wrangler` for `@prisma/adapter-d1`,
+ * which can also be a `devDependency`.
+ * Only official, well-known Driver Adapters are considered.
+ */
+export async function getPrismaDriverAdapterVersionsFromLocalPackageJson(
+  cwd: string = process.cwd(),
+): Promise<Array<DriverAdapterVersion> | null> {
+  cwd = cwd ?? process.cwd()
+  try {
+    const pkgJsonPath = await pkgUp({ cwd })
+
+    if (!pkgJsonPath) {
+      return null
+    }
+
+    const pkgJsonString = await fs.promises.readFile(pkgJsonPath, 'utf-8')
+    const pkgJson = JSON.parse(pkgJsonString) as {
+      dependencies?: Record<string, string>
+      devDependencies?: Record<string, string>
+    }
+
+    const adapterVersions = Object.entries((pkgJson.dependencies ?? {}) as Record<string, string>)
+      // Find all known Driver Adapter dependencies, usually `@prisma/adapter-*`:
+      .filter(([key]) => isOfficialDriverAdapter(key)) as Array<[OfficialDriverAdapters, string]>
+
+    const adapterVersionsWithUnderlyingDrivers = adapterVersions.map(([adapterName, adapterVersion]) => {
+      const underlyingDriver = underlyingDriverAdaptersMap[adapterName]
+      let underlyingDriverVersion = pkgJson.dependencies?.[underlyingDriver] ?? null
```

Comment:
So, that potentially will give us a version range, not the actual version: (`^5.0.0` can really be any version of Prisma 5). Can we try to require (`const require = createRequire(pkgJsonPath); require(adapterName).version` the package and fallback onto the range only if that fails?

Reply:
Wouldn't the behavior you're suggesting be different than what `prisma version` currently does?
I agree that having the precise version would be better, but if this is a new kind of logic, perhaps that should be applied in a different PR (?)

Reply:
Is it different? Engines and prisma cli version are known to cli for sure and for `@prisma/client` version we actually try to require it in a similar way I am suggestion